src/openrct2/interface/Screenshot.cpp

```diff
@@ -207,106 +212,40 @@ std::string screenshot_dump_png_32bpp(int32_t width, int32_t height, const void*
-                    return { upperX, lower };
-                case 1:
-                    return { upperX, upperY };
-                case 2:
-                    return { lower, upperY };
-                case 3:
-                    return { lower, lower };
-            }
-        case EdgeType::TOP:
-            switch (rotation)
-            {
-                default:
-                case 0:
-                    return { lower, lower };
-                case 1:
-                    return { upperX, lower };
-                case 2:
-                    return { upperX, upperY };
-                case 3:
-                    return { lower, upperY };
-            }
-        case EdgeType::RIGHT:
-            switch (rotation)
-            {
-                default:
-                case 0:
-                    return { lower, upperY };
-                case 1:
-                    return { lower, lower };
-                case 2:
-                    return { upperX, lower };
-                case 3:
-                    return { upperX, upperY };
-            }
-        case EdgeType::BOTTOM:
-            switch (rotation)
-            {
-                default:
-                case 0:
-                    return { upperX, upperY };
-                case 1:
-                    return { lower, upperY };
-                case 2:
-                    return { lower, lower };
-                case 3:
-                    return { upperY, lower };
-            }
-    }
-}
-
-static int32_t GetHighestBaseClearanceZ(const CoordsXY& location)
+static int32_t GetHighestBaseClearanceZ(const CoordsXY& location, const bool useViewClipping)
 {
     int32_t z = 0;
     auto element = map_get_first_element_at(location);
     if (element != nullptr)
     {
         do
         {
+            if (useViewClipping && (element->GetBaseZ() > gClipHeight * COORDS_Z_STEP))
+            {
+                continue;
+            }
             z = std::max<int32_t>(z, element->GetBaseZ());
             z = std::max<int32_t>(z, element->GetClearanceZ());
         } while (!(element++)->IsLastForTile());
     }
     return z;
 }
 
-static int32_t GetTallestVisibleTileTop(const TileCoordsXY& mapSize, int32_t rotation)
+static int32_t GetTallestVisibleTileTop(
+    int32_t rotation, TileCoordsXY startCoords, TileCoordsXY endCoords, const bool useViewClipping)
 {
-    int32_t minViewY = 0;
-    for (int32_t y = 1; y < mapSize.y - 1; y++)
+    int32_t minViewY = INT32_MAX;
+    for (int32_t y = startCoords.y; y <= endCoords.y; y++)
     {
-        for (int32_t x = 1; x < mapSize.x - 1; x++)
+        for (int32_t x = startCoords.x; x <= endCoords.x; x++)
         {
             auto location = TileCoordsXY(x, y).ToCoordsXY();
-            int32_t z = GetHighestBaseClearanceZ(location);
+            int32_t z = GetHighestBaseClearanceZ(location, useViewClipping);
             int32_t viewY = translate_3d_to_2d_with_z(rotation, CoordsXYZ(location, z)).y;
```

Comment:
`location` has a bit of an offset, since it's from `TileCoordsXY`, it points at the corner. Might want to get the tile center here instead, and subtract 16 from y to cover the full top of the clearance for the tile. This would explain why get different offsets from different rotations. There's a helper function to get the tile center.

Reply:
Implemented ToTileCentre helper function, this now has fixed that offset issue regarding different rotations